src/CausalModels.jl

```
@@ -1,9 +1,112 @@
 module CausalModels
-
 using Distributions
 using Statistics
-export ExogenousVariable, EndogenousVariable, intervene, randomsample, prob
+export ExogenousVariable, EndogenousVariable, randomsample, prob, intervene
+
+"""
+Struct for an exogenous variable.
+"""
+struct ExogenousVariable
+    name
+    distribution
+end
 
-# Fill in the rest
+"""
+Struct for an endogenous variable.
+"""
+struct EndogenousVariable
+    operator
+    parents
+end
+
+"""
+Evalutes an endogenous variable given values of parent exogenous variables.
+"""
+function (u::EndogenousVariable)(u_)
+    evaluated_vars = map(parent -> (parent)(u_), u.parents)
+    return reduce(u.operator, evaluated_vars)
+end
 
+"""
+Finds exogenous variable value from list of exogenous variable values.
+"""
+function (u::ExogenousVariable)(u_)
+    for name in keys(u_)
+        if u.name == name
+            return getindex(u_, name)
+        end
+    end
 end
```

What will happen if `u.name` not in `keys(u_)` (silent error)?